src/rpc/mempool.cpp

```diff
@@ -84,7 +84,7 @@ static RPCHelpMan sendrawtransaction()
                                                      DEFAULT_MAX_RAW_TX_FEE_RATE :
                                                      CFeeRate(AmountFromValue(request.params[1]));
 
-            int64_t virtual_size = GetVirtualTransactionSize(*tx);
+            int64_t virtual_size = GetVirtualTransactionSize(*tx, 0, 0 /* BUG */);
```

Comment:
Perhaps the way to fix this is to overload `BroadcastTransaction()` to accept either a `CFeeRate`, `CAmount`, or nothing. After the test-accept of the transaction, you can query `MempoolAcceptResult::m_vsize` to check the feerate.

Reply:
@glozow Isn't that exactly the behaviour that we refactored out of (to checking locally here)?

Reply:
Ah, I see what you mean. Fixed.